geode-core/src/test/java/org/apache/geode/internal/cache/ProxyBucketRegionTest.java

```diff
@@ -32,4 +46,54 @@ public void shouldBeMockable() throws Exception {
 
     assertThat(mockProxyBucketRegion.getBucketAdvisor()).isSameAs(mockBucketAdvisor);
   }
+
+  @Test
+  public void testRecoverFromDisk() throws Exception {
+    PartitionedRegion partitionedRegion = mock(PartitionedRegion.class);
+    InternalRegionArguments internalRegionArguments = mock(InternalRegionArguments.class);
+    RegionAdvisor regionAdvisor = mock(RegionAdvisor.class);
+    PartitionAttributes partitionAttributes = mock(PartitionAttributes.class);
+    InternalCache cache = mock(InternalCache.class);
+    InternalDistributedSystem ids = mock(InternalDistributedSystem.class);
+    DataPolicy dp = mock(DataPolicy.class);
+    RegionAttributes ra = mock(RegionAttributes.class);
+    DiskStoreImpl ds = mock(DiskStoreImpl.class);
+    DiskInitFile dif = mock(DiskInitFile.class);
+    DiskRegion dr = mock(DiskRegion.class);
+    DistributionManager dm = mock(DistributionManager.class);
+    DistributionConfig config = mock(DistributionConfig.class);
+    CancelCriterion cancel = mock(CancelCriterion.class);
+
+    when(internalRegionArguments.getPartitionedRegionAdvisor()).thenReturn(regionAdvisor);
+    when(regionAdvisor.getPartitionedRegion()).thenReturn(partitionedRegion);
+    when(partitionedRegion.getPartitionAttributes()).thenReturn(partitionAttributes);
+    when(partitionAttributes.getColocatedWith()).thenReturn(null);
+    when(partitionedRegion.getCache()).thenReturn(cache);
+    when(partitionedRegion.getGemFireCache()).thenReturn(cache);
+    when(cache.getInternalDistributedSystem()).thenReturn(ids);
+    when(ids.getDistributionManager()).thenReturn(dm);
+    when(partitionedRegion.getDataPolicy()).thenReturn(dp);
+    when(dp.withPersistence()).thenReturn(true);
+    when(cache.getInternalDistributedSystem()).thenReturn(ids);
+    when(partitionedRegion.getAttributes()).thenReturn(ra);
+    when(ra.getEvictionAttributes()).thenReturn(null);
+    when(partitionedRegion.getDiskStore()).thenReturn(ds);
+    when(ds.getDiskInitFile()).thenReturn(dif);
+    when(dif.createDiskRegion(any(), anyString(), anyBoolean(), anyBoolean(), anyBoolean(),
+        anyBoolean(), any(), any(), any(), any(), any(), anyString(), anyInt(), any(),
+        anyBoolean())).thenReturn(dr);
+    when(dm.getConfig()).thenReturn(config);
+    when(config.getAckWaitThreshold()).thenReturn(10);
+    when(cache.getCancelCriterion()).thenReturn(cancel);
+    when(regionAdvisor.isInitialized()).thenReturn(true);
+
+    ProxyBucketRegion proxyBucketRegion =
+        new ProxyBucketRegion(0, partitionedRegion, internalRegionArguments);
+
+    proxyBucketRegion.recoverFromDisk();
+    proxyBucketRegion.recoverFromDisk();
+    verify(regionAdvisor, times(1))
+        .isInitialized();
```

Comment:
It's not at all clear to me what this is verifying, and this test passes even with the changes in `ProxyBucketRegion` reverted, so it doesn't seem to be testing the new behaviour. I feel like for this change it might be good to add an integration test that shows the changed behaviour at a higher level, by actually creating a colocated region and showing that already recovered buckets are not recovered again.